Loop/Managers/TestingScenariosManager.swift

```diff
@@ -320,21 +322,37 @@ extension TestingScenariosManager {
         }
     }
     
-    private func reloadCGMManager(withIdentifier cgmManagerIdentifier: String) -> TestingCGMManager {
-        deviceManager.cgmManager = nil
-        let result = deviceManager.setupCGMManager(withIdentifier: cgmManagerIdentifier, prefersToSkipUserInteraction: true)
-        switch result {
-        case .success(let setupUIResult):
-            switch setupUIResult {
-            case .createdAndOnboarded(let cgmManager):
-                let cgmManager = cgmManager as! TestingCGMManager
-                cgmManager.autoStartTrace = false
-                return cgmManager
+    func reloadCGMManager(withIdentifier cgmManagerIdentifier: String) async -> TestingCGMManager {
+        var cgmManager: TestingCGMManager? = nil
+        try? await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, Error>) in
```

Comment:
Don't think this needs to be throwing since the only error here is a `fatalError`. 

Would this work?
```suggestion
        await withCheckedContinuation { continuation in
```